### PR TITLE
LTD-670: Update SIEL Licence document template

### PR DESCRIPTION
**Pull request by saruniitr**
## Change description

- Add Organisation registered number
- Add Exporter phone number
- Add product part number, serial numbers and reformat the section
- Add proviso reasons if any
- Update Authorisation section depending on products covered by firearms act
- Add additional authorisation details for temporary SIELs

### CODE REVIEW

File: api/letter_templates/context_generator.py [hunk 1]
```diff
@@ -657,15 +654,11 @@ class Meta:
         ]
 
     good = GoodSerializer()
-    firearm_details = serializers.SerializerMethodField()
+    firearm_details = FirearmDetailsSerializer()
```
Comment (alixedi) [inline api/letter_templates/context_generator.py:657]: Brilliant! 👌

File: api/letter_templates/templates/letter_templates/siel.html [hunk 2]
```diff
@@ -253,18 +270,55 @@ <h1>Standard Individual Export Licence</h1>
...
+        {% for good_item in goods.approve %}
+          {% with good=good_item.good firearm_details=good_item.firearm_details %}
             <tr>
                 <td class="border-black" colspan="1">{{ forloop.counter }}</td>
-                <td class="border-black" colspan="5">{% if good.name %}{{ good.name }}<br>{% endif %}{{ good.description }}</td>
+                <td class="border-black" colspan="5">
+                  <table border="0">
+                    <tr>
+                        <td><strong>Name:</strong></td>
+                        <td>{% if good.name %}{{ good.name }}{% endif %}{{ good.description }}
```
Comment (alixedi) [inline api/letter_templates/templates/letter_templates/siel.html:281]: Do we need `{{ good.description }}` here? (Given we are putting it in the next <tr>)

File: api/letter_templates/templates/letter_templates/siel.html [hunk 3]
```diff
@@ -253,18 +270,55 @@ <h1>Standard Individual Export Licence</h1>
...
+                    {% if firearm_details and firearm_details.has_identification_markings == "Yes" %}
+                      <tr>
+                        <td><strong>Serial number:</strong></td>
+                        <td>
+                          {% for sn in firearm_details.serial_numbers %}
+                            {% if not forloop.last %}
+                              {{ sn }} - {{ forloop.counter }},<br>
+                            {% else %}
+                              {{ sn }} - {{ forloop.counter }}
+                            {% endif %}
```
Comment (alixedi) [inline api/letter_templates/templates/letter_templates/siel.html:301]: Can we consider using [join](https://docs.djangoproject.com/en/3.1/ref/templates/builtins/#join) here to dance around the if/else logic (that seems to ensure we don't get a `,` in the end).

File: api/letter_templates/templatetags/custom_tags.py [hunk 4]
```diff
@@ -97,3 +97,29 @@ def get(object_to_search, nested_properties_list):
...
+
+@register.filter()
+def remove_duplicate_provisos(goods_approved):
+    provisos = set()
+
+    for good_item in goods_approved:
+        if "proviso_reason" in good_item:
+            provisos.add(good_item["proviso_reason"])
+
+    return provisos
```
Comment (alixedi) [inline api/letter_templates/templatetags/custom_tags.py]: The following is more succinct but it is a matter of taste -

`return {good for good in goods_approved if "proviso_reason" in good}`

File: api/letter_templates/templatetags/custom_tags.py [hunk 5]
```diff
@@ -97,3 +97,29 @@ def get(object_to_search, nested_properties_list):
...
+    if goods_approved is None:
+        return covered_by_firearms_act
+
+    for good_item in goods_approved:
+        firearm_details = good_item["firearm_details"]
+        if firearm_details and firearm_details["is_covered_by_firearm_act_section_one_two_or_five"] == "Yes":
+            covered_by_firearms_act = True
+            break
+
+    return covered_by_firearms_act
```
Comment (alixedi) [inline api/letter_templates/templatetags/custom_tags.py:115]: I think the break followed by return is redundant. 

```
if goods_approved is None:
    return False

    for good_item in goods_approved:
         if good_item["firearm_details"] and firearm_details["is_covered_by_firearm_act_section_one_two_or_five"] == "Yes":
            return True
```

**Review by alixedi — APPROVED** [review]
I'll leave a few comments to you.

File: api/letter_templates/templates/letter_templates/siel.html (hunk 3 above)
Comment (saruniitr) [reply api/letter_templates/templates/letter_templates/siel.html:301]: Agree, join is better

File: api/letter_templates/templatetags/custom_tags.py (hunk 5 above)
Comment (saruniitr) [reply api/letter_templates/templatetags/custom_tags.py:115]: yes, my bad

**Pull request merged**